Display message to user when account is closed

MonilBhavsar opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
- Display success message when account is closed
- Supports Offline functionality

### Fixed Issues
<!---
Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing.
Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the issue; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/10224

### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Tap on profile
2. Go to Security > Close Account
3. Enter email
4. Click Close Account
5. Verify "Account closed successfully" message appears on Login page as in screenshot
6. Enter a new email in the login page
7. Click "Go Back" from the password page. Verify that "Account closed successfully" message is not there

**Offline mode**

1. Tap on profile
2. Go to Security > Close Account
3. Enter email
4. Close Account should be disabled


- [x] Verify that no errors appear in the JS console

### PR Review Checklist
<!--
This is a checklist for PR authors & reviewers. Please make sure to complete all tasks and check them off once you do, or else Expensify has the right not to merge your PR!
-->
#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Tap on profile
2. Go to Security > Close Account
3. Enter email
4. Click Close Account
5. Verify "Account closed successfully" message appears on Login page as in screenshot
6. Enter a new email in the login page
7. Click "Go Back" from the password page. Verify that "Account closed successfully" message is not there

**Offline mode**

1. Tap on profile
2. Go to Security > Close Account
3. Enter email
4. Close Account should be disabled

- [ ] Verify that no errors appear in the JS console

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->
<img width="1792" alt="Screenshot 2022-10-10 at 8 52 45 PM" src="https://user-images.githubusercontent.com/32012005/194904720-bf206546-3a32-4896-9231-58c8858e79d4.png">
<img width="1792" alt="Screenshot 2022-10-10 at 9 12 11 PM" src="https://user-images.githubusercontent.com/32012005/194904742-12491f83-2576-4d3c-bd3e-5dd60eaf4378.png">


https://user-images.githubusercontent.com/32012005/195408730-55ca2ac6-1b2d-49a5-8944-476f0fbd08cc.mov


#### Mobile Web - Chrome
<!-- Insert screenshots of your changes on the web platform (from chrome mobile browser)-->
<img src=https://user-images.githubusercontent.com/32012005/194923959-7b5a0d78-7a29-4481-b901-70755a28cfe6.png  width=400>

#### Mobile Web - Safari
<!-- Insert screenshots of your changes on the web platform (from safari mobile browser)-->
<img src=https://user-images.githubusercontent.com/32012005/194917838-b4048b46-c921-493d-8754-4ad32b951f5c.png  width=400>
<img src=https://user-images.githubusercontent.com/32012005/194917844-3cba82b6-183d-4e77-bba1-163d776b5c73.png  width=400>

#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->

<img width="1192" alt="Screenshot 2022-10-10 at 9 25 54 PM" src="https://user-images.githubusercontent.com/32012005/194908048-b9fa0154-9220-4739-921d-bdb8e61313b4.png">
<img width="1192" alt="Screenshot 2022-10-10 at 9 29 59 PM" src="https://user-images.githubusercontent.com/32012005/194908056-f42e0ca7-6b51-45a9-be7d-a5d8b5ff1165.png">


#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->
<img src=https://user-images.githubusercontent.com/32012005/194917882-2346ff45-d8ed-4c14-ac99-3b31c9928fac.png  width=400>
<img src=https://user-images.githubusercontent.com/32012005/194917887-a4c38ef5-f13c-4489-a1cf-388e4e1d52cb.png  width=400>

#### Android
<!-- Insert screenshots of your changes on the Android platform-->
<img src=https://user-images.githubusercontent.com/32012005/194923895-34eef880-593c-4e92-8ac9-6d5210e5d20f.png  width=400>
<img src=https://user-images.githubusercontent.com/32012005/194923898-c710f487-f352-41e0-8b0b-a81de1da160f.png  width=400>

cc @Expensify/design

MonilBhavsar committed:
Display message to user if action succeeded

MonilBhavsar committed:
Use closeAccount Onyx key

MonilBhavsar committed:
Add offline support and offline indicator

Beamanator reviewed (changes requested):
So far looking great! A few comments, and also it looks like we don't clear the success message in this flow but maybe should:
1. Log in with account, close it
2. Verify you're back on the login page, and you see the close success message
3. Enter another account's email address, click "Continue"
4. Don't sign in, then click "Go back"
5. You now still see the "close account success message", but I don't think you should

Julesssss reviewed (commented):
The code looks good, and the first test passed. But confusingly I'm not getting the offline state on web 😕 

<img width="1792" alt="Screenshot 2022-10-11 at 14 28 53" src="https://user-images.githubusercontent.com/10736861/195105227-6defadaf-f2a7-4ba2-9ec8-aca83d10b68b.png">

shawnborton commented on src/pages/settings/Security/CloseAccountPage.js:
```diff
@@ -110,9 +116,11 @@ class CloseAccountPage extends Component {
+                        isDisabled={Str.removeSMSDomain(userEmailOrPhone).toLowerCase() !== this.state.phoneOrEmail.toLowerCase() || this.props.network.isOffline}
                         style={[styles.mt5]}
                     />
+                    {!this.props.isSmallScreenWidth
+                        && <OfflineIndicator containerStyles={[styles.mt2]} />}
```
What exactly do you mean by that Alex? Just want to make sure I understand as a non-engineer 😄

Beamanator replied:
Yeah good question, so my train of thought is:
- If we're adding the Offline Indicator to the "Close Account" page, why wasn't it there before?
- Then I checked if the Offline Indicator is present on other form pages (example: the Profile Page) and I don't see the offline indicator there
- It seems weird to me to show the user they're offline in the Close Account page w/out also showing they're offline in the Profile page, so I'm basically asking here if we want to show the Offline Indicator on **all** pages, or only on pages with forms, or something else

One other idea I had is: In web / desktop full screen mode, it may not be necessary to show the Offline Indicator twice (you can see that it's already showing in the chat report page) so maybe we **only** show the Offline Indicator on sub-pages on small screens (including mobile devices)?

Julesssss replied:
> It seems weird to me to show the user they're offline in the Close Account page w/out also showing they're offline in the Profile page, so I'm basically asking here if we want to show the Offline Indicator on all pages, or only on pages with forms, or something else

Maybe I misunderstood. But aren't these different offline patterns? We let users modify their profiles offline, but we shouldn't let users delete accounts while offline.

Beamanator replied:
Hmm true true, but doesn't it feel a bit weird seeing an offline status in some pages and not in all?

Julesssss replied:
I don't think so. By that logic we would be forced to make all settings offline OR non-offline. I don't think we should care about a consistent offline behavior on a feature that lets users stop being our customer

Beamanator replied:
IMO, the message `You appear to be offline` is good to know everywhere or nowhere, not "at the bottom of some page but not all".

> I don't think we should care about a consistent offline behavior **on a feature that lets users stop being our customer**

I'm a bit confused how this would let users stop being our customer, do you mean if users see `You appear to be offline` they'll assume the form they're seeing won't be submittable, so they'll think our app doesn't work offline even though some pages will and some won't?

Julesssss replied:
OH. Whoops.

I was mistaken and thought this was regarding disabling the Close Account button while offline... not the indicator. Ignore me

Beamanator replied:
Aahhhh ok that makes more sense @Julesssss 😅 

> May be worth a discussion in channel, why it was only implemented for mobile and not web and then we can implement it.

Ooh that's a good idea 👍

Julesssss replied:
Shall we leave that discussion for now? I think we could merge this today and revisit.

> why it was only implemented for mobile and not web and then we can implement it.

As for this, I think it works now because the page is shown as a RightDockedModal, right?

MonilBhavsar committed:
Address reviews and clear success message

MonilBhavsar commented:
This is ready for a re review. Updated testing and QA steps and added screen recording of the edge case @Beamanator pointed out. Thanks for pointing it out and helping with it!

MonilBhavsar commented:
> The code looks good, and the first test passed. But confusingly I'm not getting the offline state on web 😕

@Julesssss hope this was not consistent and something flakey with network?

Julesssss commented:
Hey @MonilBhavsar, actually yeah. I'm pretty sure now that this is an unrelated [issue on my side](https://github.com/Expensify/App/issues/11783#issuecomment-1277447112). But for that reason, I'm unable to verify the tests 😞

Julesssss reviewed (commented):
It looks like we'll take the discussion to Slack. But once that is resolved I'll be happy to retest on iOS/Android/mWeb. Unfortunately someone else will need to verify web, due to my current [issue](https://github.com/Expensify/App/issues/11783#issuecomment-1277447112).

techievivek commented:
I unassigned myself because I won't be able to review it by tomorrow. If this continues for next week, I will jump in again 😄

MonilBhavsar commented:
Off hold

Beamanator commented:
Honestly I'm having lotsss of trouble with my VM today so I think it's up to you @Julesssss 😅

Julesssss commented:
Okay, offline functionality is finally working for me. Tested on all platforms: 

Desktop
<img width="1007" alt="Screenshot 2022-10-14 at 14 23 09" src="https://user-images.githubusercontent.com/10736861/195864229-2826d110-c98c-4164-a11c-b085e253b7dd.png">

Web
<img width="1216" alt="Screenshot 2022-10-14 at 14 17 18" src="https://user-images.githubusercontent.com/10736861/195864233-0000c89d-c41e-4bd3-afba-60125dd9fad7.png">

Android
![Screenshot_1665754421](https://user-images.githubusercontent.com/10736861/195864625-d6f9e31b-86a4-430d-9186-275349c0a1da.png)
![Screenshot_1665754517](https://user-images.githubusercontent.com/10736861/195864659-c686ba0d-85ec-4612-8492-7bf589b2af14.png)

iOS
![Simulator Screen Shot - iPhone 13 - 2022-10-14 at 15 04 43](https://user-images.githubusercontent.com/10736861/195866717-237e0667-6262-4265-91e0-1cde26435173.png)

Julesssss reviewed (approved):
Tested successfully on all platforms ✅

Julesssss commented:
PR Reviewer Checklist 
- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

MonilBhavsar commented:
Thank you @Julesssss for testing and reviewing!
@Beamanator all yours

Beamanator commented:
Sorry all, was battling VM issues the last few days, but I think I'm back and ready to review!

Beamanator reviewed (approved):
:shipit:

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

Beamanator commented:
I'm 99.9% sure I saw all tests were passing before merging 🤔

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/3281110796) to staging by @Beamanator in version: 1.2.18-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/3316882976) to production by @chiragsalian in version: 1.2.18-10 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

Gonals commented:
Heads up @MonilBhavsar  @Beamanator  @Julesssss ! This PR caused a bug here: https://github.com/Expensify/App/issues/13506

Not a big deal, though. I think is more a change in approach than a bug.

Beamanator commented:
Yeahhhh I agree it's more of a change in approach - I think I argued that it was better to keep the closed account message for longer, but no big arguments / opinions either way :D 👍
